pendingBlock.go

```diff
@@ -153,35 +157,41 @@ func (b *pendingBlock) nextTransaction() *flowgo.TransactionBody {
 // This function uses the provided execute function to perform the actual
 // execution, then updates the pending block with the output.
 func (b *pendingBlock) ExecuteNextTransaction(
-	execute func(ledgerView state.View, txIndex uint32, tx *flowgo.TransactionBody) (*fvm.TransactionProcedure, error),
-) (*fvm.TransactionProcedure, error) {
-	tx := b.nextTransaction()
-
-	childView := b.ledgerView.NewChild()
+	vm *fvm.VirtualMachine,
+	ctx fvm.Context,
+) (
+	fvm.ProcedureOutput,
+	error,
+) {
+	txnBody := b.NextTransaction()
+	txnIndex := b.index
 
 	// increment transaction index even if transaction reverts
 	b.index++
 
-	tp, err := execute(childView, b.index, tx)
```

Comment:
Note: This was an off-by-one error.  transaction index should start 0 instead of 1.